lib/beaker/dsl/helpers/facter_helpers.rb

```diff
@@ -42,6 +42,24 @@ def fact(name, opts = {})
           fact_on(default, name, opts)
         end
 
+        # Create a custom fact on the provided host
+        # We do this by setting a static value in the facts.d directory
+        #
+        # @param [Host, Array<Host>, String, Symbol] host    One or more hosts to act upon,
+        #                            or a role (String or Symbol) that identifies one or more hosts.
+        # @param [String] name The name of the fact to create
+        # @param [String, Array] value  The value of the fact to create
+        def create_fact_on(host, name, value)
+          block_on host do | host |
+            create_remote_file(host, File.join(host.puppet['facter-factsd'], "#{name}.yaml"), value.to_yaml)
```

Comment:
@electrical, looking through this, I'm seeing that you're using `host.puppet['facter-factsd']` in order to get the paths.  That call won't work for paths defined in the host defaults, that tries to get the values from a puppet configprint call.  In order to get the values you've defined, you'll have to change that to `host['facter-factsd']`.